src/libraries/Common/src/Interop/Windows/Kernel32/Interop.FindFirstFileEx.cs

```diff
@@ -24,6 +24,21 @@ internal static SafeFindHandle FindFirstFile(string fileName, ref WIN32_FIND_DAT
             return FindFirstFileExPrivate(fileName, FINDEX_INFO_LEVELS.FindExInfoBasic, ref data, FINDEX_SEARCH_OPS.FindExSearchNameMatch, IntPtr.Zero, 0);
         }
 
+        internal static void GetFindData(string fullPath, bool isDirectory, bool ignoreAccessDenied, ref WIN32_FIND_DATA findData)
+        {
+            using SafeFindHandle handle = FindFirstFile(Path.TrimEndingDirectorySeparator(fullPath), ref findData);
+            if (handle.IsInvalid)
+            {
+                int errorCode = Marshal.GetLastPInvokeError();
+                // File not found doesn't make much sense coming from a directory.
+                if (isDirectory && errorCode == Errors.ERROR_FILE_NOT_FOUND)
+                    errorCode = Errors.ERROR_PATH_NOT_FOUND;
+                if (ignoreAccessDenied && errorCode == Errors.ERROR_ACCESS_DENIED)
+                    return;
+                throw Win32Marshal.GetExceptionForWin32Error(errorCode, fullPath);
+            }
+        }
```

Comment:
nit: we already have a `GetFindData` method, would be nice to re-use this one there.
https://github.com/dotnet/runtime/blob/e8a18d691c4316eb20e22c00158eac5d98cee987/src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs#L276-L289